### PR TITLE
Lint all Markdown files and enable CI MD lint

### PR DESCRIPTION
**Pull request by ben-alkov**
Signed-off-by: Ben Alkov <ben.alkov@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [n/a] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request includes link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,57 +2,95 @@
...
-* Code should be readable, meaning:
-  * Comment where appropriate
-  * Separate logical parts of methods/functions by newline
-  * Use meaningful names for methods/functions/classes/variables
+- Maximum line length is 100 characters
+- Pull Request *must* pass following checks
+  - [Travis CI test][]
+    - This means that no tests fail on Python 2.7 and >= 3.4
+    - You can run tests locally using `pytest`
+  - [Landscape test][]
```
Comment (MartinBasti) [inline CONTRIBUTING.md:64]: This is dead, it may be removed in another PR

Comment (ben-alkov) [reply CONTRIBUTING.md:64]: Yeah, I left the *content* completely untouched (and if I didn't, it's a bug :) )...

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +39,10 @@ env:
       OS_VERSION=31
       PYTHON_VERSION=3
       ENGINE=docker
+    - ACTION=markdownlint
+      CONTAINER_NAME=ruby
```
Comment (MartinBasti) [inline .travis.yml:43]: I don't think that container name should be defined here. It may break local testing due conflicting names

File: test.sh [hunk 3]
```diff
@@ -7,45 +7,21 @@ OS=${OS:="centos"}
 OS_VERSION=${OS_VERSION:="7"}
 PYTHON_VERSION=${PYTHON_VERSION:="2"}
 ACTION=${ACTION:="test"}
-IMAGE="$OS:$OS_VERSION"
-
-# Optionally specify repo and branch for osbs-client to test changes
-# which depend on osbs-client patches not yet available in upstream master
-OSBS_CLIENT_REPO=${OSBS_CLIENT_REPO:-https://github.com/containerbuildsystem/osbs-client}
-OSBS_CLIENT_BRANCH=${OSBS_CLIENT_BRANCH:-master}
+IMAGE=${IMAGE:="$OS:$OS_VERSION"}
```
Comment (MartinBasti) [inline test.sh:10]: This should be dynamic, i.e. use ruby for mdlint here

File: test.sh [hunk 4]
```diff
@@ -7,45 +7,21 @@ OS=${OS:="centos"}
...
+# The following might be referenced outside of `prep_linux`, i.e. before they're defined
+CONTAINER_NAME=${CONTAINER_NAME:="atomic-reactor-$OS-$OS_VERSION-py$PYTHON_VERSION"}
```
Comment (MartinBasti) [inline test.sh:12]: This should be generated dynamic, i.e just `atomic-reactor-markdownlint-ruby` for makrdown lint. ALso fix container anme for bandit and pylint, it uses conflicting names

File: test.sh [hunk 5]
```diff
@@ -54,103 +30,140 @@ elif [[ $($ENGINE ps -q -f name="$CONTAINER_NAME" | wc -l) -eq 0 ]]; then
   $ENGINE container start "$CONTAINER_NAME"
 fi
 
-if [[ $OS == "centos" ]]; then
-  # Don't let builddep enable *-source repos since they give 404 errors
-  $RUN rm -f /etc/yum.repos.d/CentOS-Sources.repo
-fi
+function prep_linux() {
```
Comment (MartinBasti) [inline test.sh:33]: maybe install_osbs ?

File: test.sh [hunk 6]
```diff
@@ -54,103 +30,140 @@ elif [[ $($ENGINE ps -q -f name="$CONTAINER_NAME" | wc -l) -eq 0 ]]; then
...
+  ;;
 *)
   echo "Unknown action: ${ACTION}"
   exit 2
   ;;
 esac
 
 # Run tests
-$RUN ${TEST_CMD} "$@"
+$RUN "${TEST_CMD}" "$@"
```
Comment (MartinBasti) [inline test.sh:166]: Why this change?

P.S. it's breaking test execution

Comment (ben-alkov) [reply test.sh:166]: Lame attempt to fix quoting issue

File: test.sh (hunk 3 above)
Comment (MartinBasti) [reply test.sh:10]: I don't want to guess which image to run when I want to run linter locally. I want just specify action and enjoying output. IMAGE shall be specified in test.sh for markdownlint

Comment (ben-alkov) [reply test.sh:10]: I see, OK.
